lib/web/apiserver.go

```diff
@@ -2355,6 +2360,132 @@ func createIdentityContext(login string, sessionCtx *SessionContext) (srv.Identi
 	}, nil
 }
 
+type UILock struct {
+	Name      string           `json:"name"`
+	Message   string           `json:"message"`
+	Expires   string           `json:"expires"`
+	CreatedAt string           `json:"createdAt"`
+	CreatedBy string           `json:"createdBy"`
+	Targets   types.LockTarget `json:"targets"`
+}
+
+func (h *Handler) getClusterLocks(
+	w http.ResponseWriter,
+	r *http.Request,
+	p httprouter.Params,
+	sessionCtx *SessionContext,
+	site reversetunnel.RemoteSite) (interface{}, error) {
+	ctx := r.Context()
+	clt, err := sessionCtx.GetUserClient(r.Context(), site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	locks, err := clt.GetLocks(ctx, false)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// Lock data structure is reformatted to save doing it on the client as the
+	// Table component doesn't support nested complex objects. And fails to properly
+	// sort or filter results.
+	lockList := make([]UILock, 0, len(locks))
+	for i := 0; i < len(locks); i++ {
+		expires := ""
+		createdAt := ""
+		if locks[i].LockExpiry() != nil {
+			expires = locks[i].LockExpiry().Format(time.RFC3339Nano)
+		}
+
+		if !locks[i].CreatedAt().IsZero() {
+			createdAt = locks[i].CreatedAt().Format(time.RFC3339Nano)
+		}
+
+		lockList = append(lockList, UILock{
+			Name:      locks[i].GetMetadata().Name,
+			Message:   locks[i].Message(),
+			Expires:   expires,
+			Targets:   locks[i].Target(),
+			CreatedAt: createdAt,
+			CreatedBy: locks[i].CreatedBy(),
+		})
+	}
+	return lockList, nil
+}
+
+type createLockReq struct {
+	Targets types.LockTarget `json:"targets"`
+	Message string           `json:"message"`
+	TTL     string           `json:"ttl"`
+}
+
+func (h *Handler) createClusterLock(
+	w http.ResponseWriter,
+	r *http.Request,
+	p httprouter.Params,
+	sessionCtx *SessionContext,
+	site reversetunnel.RemoteSite) (interface{}, error) {
+	var req *createLockReq
+
+	if err := httplib.ReadJSON(r, &req); err != nil {
```

Comment:
```suggestion
	var req *createLockReq
	if err := httplib.ReadJSON(r, &req); err != nil {
```

I'd remove this empty line, the req is closely related to the following line which populates it.